src/graph.cpp

```diff
@@ -162,18 +166,24 @@ int Graph::addToTour(std::shared_ptr<Vertex> stop) {
  * @brief Iterates through the vertex set using DFS, respecting if an edge is selected or not
  * Time Complexity: O(|V|+|E|)
  * @param source - Vertex where the DFS starts
+ * @return execution errors (0 if none, -1 if couldn't calculate Edge length, -2 if self-loop)
  */
-void Graph::preorderMSTTraversal(std::shared_ptr<Vertex> source){
-/*    source->setVisited(true);
-    addToTour(source);
-    for (std::shared_ptr<Edge> &e : source->getAdj()){
-        if(!(e->getDest()->isVisited()) && e->isSelected()){
-            preorderMSTTraversal(e->getDest());
+int Graph::preorderMSTTraversal(std::shared_ptr<Vertex> source){
+    source->setVisited(true);
+    int exec_val = addToTour(source);
+    if (exec_val != 0) return exec_val;
+
+    for (size_t i = 0; i<vertexSet.size(); i++){
+        std::shared_ptr<Vertex> dest = findVertex(i);
+
+        if(!(dest->isVisited()) && selectedEdges[source->getId()][i]){
+            preorderMSTTraversal(dest);
         }
     }
 
-    if (source->getId() == 0) addToTour(source);
-*/}
+    if (source->getId() == 0) return addToTour(source);
+    return 0;
+}
 
 /**
  * Calculates an approximation of the TSP, using the triangular approximation heuristic
```

Comment:
This complexity here in theory is O(V + E), I am not sure in our case it would be O(V + E + E*logV) so O(V + E(logV + 1)) so perhaps just O(V + E*logV), I don't know, let me know what you think

Reply:
It's O(V^2) as this function only calls other functions and the max complexity of the other functions is O(V^2)